core/src/main/java/org/elasticsearch/action/update/UpdateResponse.java

```diff
@@ -109,4 +117,65 @@ public String toString() {
         builder.append(",shards=").append(getShardInfo());
         return builder.append("]").toString();
     }
+
+
+    /**
+     * ConstructingObjectParser used to parse the {@link UpdateResponse}. We use a ObjectParser here
+     * because most fields are parsed by the parent abstract class {@link DocWriteResponse} and it's
+     * not easy to parse part of the fields in the parent class and other fields in the children class
+     * using the usual streamed parsing method.
+     */
```

Comment:
I think you can remove this comment ;) no need to explain why you are using ObjectParser I think :)